src/Microsoft.DotNet.Tools.Repl.Csi/Program.cs

```diff
@@ -19,21 +24,128 @@ public static int Main(string[] args)
+                                .Execute();
+        }
+
+        private static List<string> GetRuntimeDependencies(ProjectContext projectContext, string buildConfiguration)
+        {
+            var runtimeDependencies = new List<string>();
+
+            var projectExporter = projectContext.CreateExporter(buildConfiguration);
+            var projectDependencies = projectExporter.GetDependencies();
+
+            foreach (var projectDependency in projectDependencies)
+            {
+                var runtimeAssemblies = projectDependency.RuntimeAssemblies;
+
+                foreach (var runtimeAssembly in runtimeAssemblies)
+                {
+                    var runtimeAssemblyPath = runtimeAssembly.ResolvedPath;
+
+                    if (!runtimeDependencies.Contains(runtimeAssemblyPath))
+                    {
+                        runtimeDependencies.Add(runtimeAssemblyPath);
+                    }
+                }
+            }
+
+            return runtimeDependencies;
+        }
+
+        private static string CreateResponseFile(ProjectContext projectContext, string buildConfiguration, string tempOutputDir)
+        {
+            var outputFileName = projectContext.ProjectFile.Name;
+            var outputFilePath = Path.Combine(tempOutputDir, $"{outputFileName}{Constants.DynamicLibSuffix}");
+            var projectResponseFilePath = Path.Combine(tempOutputDir, $"dotnet-repl.{outputFileName}{Constants.ResponseFileSuffix}");
+
+            var runtimeDependencies = GetRuntimeDependencies(projectContext, buildConfiguration);
+
+            using (var fileStream = new FileStream(projectResponseFilePath, FileMode.Create))
+            {
+                var streamWriter = new StreamWriter(fileStream);
+
+                streamWriter.WriteLine($"/r:\"{outputFilePath}\"");
+
+                foreach (var projectDependency in runtimeDependencies)
+                {
+                    streamWriter.WriteLine($"/r:\"{projectDependency}\"");
+                }
+
+                streamWriter.Flush();
+            }
+
+            return projectResponseFilePath;
+        }
+
+        private static int Run(string script, IEnumerable<string> targetFrameworks, string buildConfiguration, bool preserveTemporaryOutput, string projectPath)
         {
             var corerun = Path.Combine(AppContext.BaseDirectory, Constants.HostExecutableName);
             var csiExe = Path.Combine(AppContext.BaseDirectory, "csi.exe");
-            var csiArgs = string.IsNullOrEmpty(scriptOpt) ? "-i" : scriptOpt;
-            var result = Command.Create(csiExe, csiArgs)
+            var csiArgs = new StringBuilder();
+
+            var tempOutputDir = string.Empty;
+
+            if (!string.IsNullOrWhiteSpace(projectPath))
+            {
+                var projectContext = GetProjectContext(targetFrameworks, projectPath);
+
+                var compileResult = CompileProject(projectContext, buildConfiguration, out tempOutputDir);
```

Comment:
If the framework from the command-line is not in `project.json`, `projectContext` will be null, and `CompileProject` will throw a `NullReferenceException`. Should an error be reported instead?